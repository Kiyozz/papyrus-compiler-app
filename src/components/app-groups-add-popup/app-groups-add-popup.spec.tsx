import { shallow } from 'enzyme'
import React from 'react'
import AppGroupsAddPopup from './app-groups-add-popup'

describe('<AppGroupsAddPopup>', () => {
  const render = () => {
    const callback = () => {}

    const props = {
      lastId: 0,
      onGroupAdd: callback,
      onGroupEdit: callback,
      onClose: callback
    }

    return (
      <AppGroupsAddPopup {...props} />
    )
  }

  it('renders correctly', () => {
    const component = (
      render()
    )

    expect(component).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = shallow((
      render()
    ))

    expect(component).toMatchSnapshot()
    expect(component).toContainExactlyOneMatchingElement('form')
  })
})
