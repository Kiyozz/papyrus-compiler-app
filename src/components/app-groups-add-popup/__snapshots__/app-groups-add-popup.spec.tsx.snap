// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppGroupsAddPopup> matches snapshot 1`] = `
<div
  className="app-groups-add-popup"
  onClick={[Function]}
>
  <div
    className="app-groups-add-popup-container"
  >
    <form
      className="h-100 d-flex flex-column"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="group-name"
        >
          Name
        </label>
        <input
          className="form-control"
          id="group-name"
          name="group-name"
          onChange={[Function]}
          value=""
        />
      </div>
      <div
        className="form-group h-100 overflow-auto cursor-pointer"
        onBlur={[Function]}
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        title="Add scripts"
      >
        <input
          accept=".psc"
          autoComplete="off"
          multiple={true}
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          tabIndex={-1}
          type="file"
        />
        Drop your scripts files here or click here
        <CSSTransition
          classNames="app-fade"
          in={false}
          mountOnEnter={true}
          timeout={150}
          unmountOnExit={true}
        >
          <div
            className="app-groups-is-dragging-container"
          >
            Drop files here...
          </div>
        </CSSTransition>
        <div
          className="app-groups-popup-scripts"
        />
      </div>
      <div
        className="app-groups-add-popup-form-actions"
      >
        <button
          className="btn btn-primary"
          type="submit"
        >
          Add
        </button>
        <button
          className="btn btn-secondary"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
      </div>
    </form>
  </div>
</div>
`;
